diff --git a/src/pylorax/__init__.py b/src/pylorax/__init__.py
index 8700ff525..9060c1308 100644
--- a/src/pylorax/__init__.py
+++ b/src/pylorax/__init__.py
@@ -61,7 +61,7 @@
 DRACUT_DEFAULT = ["--xz", "--install", "/.buildstamp", "--no-early-microcode", "--add", "fips"]
 
 # Used for DNF conf.module_platform_id
-DEFAULT_PLATFORM_ID = "platform:f32"
+DEFAULT_PLATFORM_ID = "platform:f34"
 
 class ArchData(DataHolder):
     lib64_arches = ("x86_64", "ppc64le", "s390x", "ia64", "aarch64")
diff --git a/src/pylorax/cmdline.py b/src/pylorax/cmdline.py
index 59e851d3d..50dab7439 100644
--- a/src/pylorax/cmdline.py
+++ b/src/pylorax/cmdline.py
@@ -306,7 +306,7 @@ def lmc_parser(dracut_default=""):
 
     parser.add_argument("--project", default="Linux",
                         help="substituted for @PROJECT@ in bootloader config files")
-    parser.add_argument("--releasever", default="32",
+    parser.add_argument("--releasever", default="34",
                         help="substituted for @VERSION@ in bootloader config files")
     parser.add_argument("--volid", default=None, help="volume id")
     parser.add_argument("--squashfs-only", action="store_true", default=False,
diff --git a/src/pylorax/creator.py b/src/pylorax/creator.py
index 559912ed4..9ed3d4b5f 100644
--- a/src/pylorax/creator.py
+++ b/src/pylorax/creator.py
@@ -149,7 +149,7 @@ def dracut_args(opts):
 
 def make_appliance(disk_img, name, template, outfile, networks=None, ram=1024,
                    vcpus=1, arch=None, title="Linux", project="Linux",
-                   releasever="32"):
+                   releasever="34"):
     """
     Generate an appliance description file
 
@@ -163,7 +163,7 @@ def make_appliance(disk_img, name, template, outfile, networks=None, ram=1024,
     :param str arch: CPU architecture. Default is 'x86_64'
     :param str title: Title, passed to template. Default is 'Linux'
     :param str project: Project, passed to template. Default is 'Linux'
-    :param str releasever: Release version, passed to template. Default is 32
+    :param str releasever: Release version, passed to template. Default is 34
     """
     if not (disk_img and template and outfile):
         return None
diff --git a/src/pylorax/dnfbase.py b/src/pylorax/dnfbase.py
index c4d514f1a..de9c956a2 100644
--- a/src/pylorax/dnfbase.py
+++ b/src/pylorax/dnfbase.py
@@ -25,7 +25,7 @@
 
 def get_dnf_base_object(installroot, sources, mirrorlists=None, repos=None,
                         enablerepos=None, disablerepos=None,
-                        tempdir="/var/tmp", proxy=None, releasever="32",
+                        tempdir="/var/tmp", proxy=None, releasever="34",
                         cachedir=None, logdir=None, sslverify=True, dnfplugins=None):
     """ Create a dnf Base object and setup the repositories and installroot
 
