diff -Nuar lorax-29.16/share/templates.d/99-generic/config_files/aarch64/grub2-efi.cfg lorax-29.16-new/share/templates.d/99-generic/config_files/aarch64/grub2-efi.cfg
--- a/share/templates.d/99-generic/config_files/aarch64/grub2-efi.cfg	2019-08-26 04:14:56.212000000 -0400
+++ b/share/templates.d/99-generic/config_files/aarch64/grub2-efi.cfg	2019-08-26 04:18:31.500000000 -0400
@@ -27,15 +27,15 @@
 
 ### BEGIN /etc/grub.d/10_linux ###
 menuentry 'Install @PRODUCT@ @VERSION@ with GUI mode' --class red --class gnu-linux --class gnu --class os {
-        linux @KERNELPATH@ @ROOT@ ro inst.geoloc=0
+        linux @KERNELPATH@ @ROOT@ ro inst.geoloc=0 inst.kdump_addon=on
         initrd @INITRDPATH@
 }
 menuentry 'Install @PRODUCT@ @VERSION@ with text mode' --class red --class gnu-linux --class gnu --class os {
-        linux @KERNELPATH@ @ROOT@ ro inst.text inst.geoloc=0 console=ttyAMA0 console=tty0
+        linux @KERNELPATH@ @ROOT@ ro inst.text inst.geoloc=0 inst.kdump_addon=on console=ttyAMA0 console=tty0
         initrd @INITRDPATH@
 }
 menuentry 'Test this media & install @PRODUCT@ @VERSION@' --class red --class gnu-linux --class gnu --class os {
-	linux @KERNELPATH@ @ROOT@ rd.live.check inst.geoloc=0
+	linux @KERNELPATH@ @ROOT@ rd.live.check inst.geoloc=0 inst.kdump_addon=on
 	initrd @INITRDPATH@
 }
 menuentry 'Rescue a @PRODUCT@ system' --class red --class gnu-linux --class gnu --class os {
diff -Nuar lorax-29.16/share/templates.d/99-generic/config_files/aarch64/grub.conf lorax-29.16-new/share/templates.d/99-generic/config_files/aarch64/grub.conf
--- a/share/templates.d/99-generic/config_files/aarch64/grub.conf	2019-08-26 04:14:56.228000000 -0400
+++ b/share/templates.d/99-generic/config_files/aarch64/grub.conf	2019-08-26 04:19:04.176000000 -0400
@@ -5,9 +5,9 @@
 hiddenmenu
 title Install @PRODUCT@ @VERSION@
 	findiso
-	kernel @KERNELPATH@ @ROOT@ quiet inst.geoloc=0
+	kernel @KERNELPATH@ @ROOT@ quiet inst.geoloc=0 inst.kdump_addon=on
 	initrd @INITRDPATH@
 title Test this media & install @PRODUCT@ @VERSION@
 	findiso
-	kernel @KERNELPATH@ @ROOT@ rd.live.check quiet inst.geoloc=0
+	kernel @KERNELPATH@ @ROOT@ rd.live.check quiet inst.geoloc=0 inst.kdump_addon=on
 	initrd @INITRDPATH@
